kanidmd/src/lib/value.rs

```diff
@@ -1590,6 +1628,22 @@ mod tests {
         assert!(val3.validate());
     }
 
+    #[test]
+    fn test_value_email_address() {
+        // https://html.spec.whatwg.org/multipage/forms.html#valid-e-mail-address
+        let val1 = Value::new_email_address_s("william@blackhats.net.au");
+        let val2 = Value::new_email_address_s("alice@idm.example.com");
+        let val3 = Value::new_email_address_s("test+mailbox@foo.com");
+        let inv1 = Value::new_email_address_s("william");
+        let inv2 = Value::new_email_address_s("test~uuid");
```

Comment:
What is the `_s` suffix for? Also we can use [Django's email tests](https://github.com/django/django/blob/main/tests/validators/tests.py#L48), which is what the source code of validator uses for testing.

Reply:
takes a str ref, rather than a string. Again, something something past william, poor choices, something something.

Reply:
Given I've mentioned it a few times, opening https://github.com/kanidm/kanidm/issues/468 to address it.